Expo1900/Expo1900/Controller/TableViewController.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  TableViewController.swift
+//  Expo1900
+//
+//  Created by H on 3/12/24.
+//
+
+import UIKit
+
+class TableViewController: UITableViewController {
+
+    var list: [Item] = []
+    
+    @IBOutlet var itemsTableView: UITableView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        loadItems()
+        configureTableView()
+    }
+    
+    private func loadItems() {
+        guard let asset = NSDataAsset.init(name: "items") else {
+            return
+        }
+        
+        let decoder = JSONDecoder()
+        
+        do {
+            list = try decoder.decode([Item].self, from: asset.data)
+        } catch {
+            print("Error decoding JSON: \(error.localizedDescription)")
+        }
+    }
+    
+    private func configureTableView() {
+        itemsTableView.delegate = self
+        itemsTableView.dataSource = self
+        itemsTableView.estimatedRowHeight = 100
+        itemsTableView.rowHeight = UITableView.automaticDimension
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return list.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = itemsTableView.dequeueReusableCell(withIdentifier: "itemCell", for: indexPath) as? TableViewCell else {
+            fatalError("Error: 'itemCell' not found")
+        }
+        
+        let item: Item = list[indexPath.row]
+        
+        cell.nameLabel.text = "\(item.name)"
+        cell.shortDescriptionLabel.text = "\(item.summary)"
+        cell.itemImageView.image = UIImage(named: "\(item.imageUrl)")
+        cell.accessoryType = .disclosureIndicator
+        
+        return cell
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "listToDetail" {
+            if let cell = sender as? UITableViewCell {
+                if let indexPath = itemsTableView.indexPath(for: cell){
+                    if let vc = segue.destination as? DetailViewController {
+                        let item = list[indexPath.row]
+                        vc.item = item
+                        vc.navigationItem.title = "\(item.name)"
+                    }
+                }
+            }
+        }
+    }
```

Comment:
```suggestion
  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    guard segue.identifier == "listToDetail",
          let cell = sender as? UITableViewCell,
          let indexPath = itemsTableView.indexPath(for: cell),
          let vc = segue.destination as? DetailViewController else {
              return
          }
    
    let item = list[indexPath.row]
    vc.item = item
    vc.navigationItem.title = item.name
}
```

이렇게 작성하면 더 깔끔 할것 같아요 !
if 의 depth가 너무 깊어지면 (if문의 중첩) 가독성이 좋아지지 않아요.

Reply:
맞네요! if의 중첩을 줄이면 코드의 가독성이 더 높아지네요! 수정하였습니다 :)